source/common/http/conn_pool_grid.cc

```diff
@@ -317,6 +327,10 @@ ConnectionPool::Cancellable* ConnectivityGrid::newStream(Http::ResponseDecoder&
     // WrappedCallbacks object has also been deleted.
     return nullptr;
   }
+  if (!delay_tcp_attempt) {
+    // Immediately start TCP attempt if HTTP/3 failed recently.
+    wrapped_callbacks_.front()->tryAnotherConnection();
```

Comment:
Should we stop the failover timer at this point since there is nothing else to try?